src/usb_moded-network.c

```diff
@@ -299,18 +300,16 @@ static int resolv_conf_dns(struct ipforward_data *ipforward)
 
 static int checklink(void)
 {
-   char *dest = malloc( 32 * sizeof(char)) ;
-   int ret = 0;
-
-   readlink("/etc/udhcpd.conf", dest, 32);
-   if(dest != NULL)
-   {
-	strcpy(&dest[31], "\0");
-	ret = (strcmp(dest, UDHCP_CONFIG_PATH));
-   }
-   free(dest);
-
-   return(ret);
+  int ret = -1;
+  const size_t bufsiz = 32;
+  char *dest = malloc(bufsiz);
```

Comment:
Why do a heap allocation at all? A stack-allocated buffer of size `PATH_MAX` should do the trick?

```
char dest[PATH_MAX];

size_t len = readlink(UDHCP_CONFIG_LINK, dest, sizeof(dest));

if (len == -1) {
    return -1;
}

dest[len] = 0;
return strcmp(dest, UDHCP_CONFIG_PATH);
```